File: leetcode/125. Valid Palindrome.ts
//https://leetcode.com/problems/valid-palindrome/description/
function isPalindrome(s: string): boolean {
    const arrStr: string[] = []

    for(let w of s) {
        if(w.match(/[A-Za-z0-9]/i)) arrStr.push(w.toLowerCase())
    }

    for(let i = 0; i < arrStr.length; i++) {
        if(arrStr[i] !== arrStr[arrStr.length -1 - i]) return false
    }

    return true
};

File: leetcode/125. Valid Palindrome.go
import "unicode"

// https://leetcode.com/problems/valid-palindrome/description/
func isPalindrome(s string) bool {
	arr := []rune{}
	for _, v := range s {
		if unicode.IsLetter(v) || unicode.IsDigit(v) {
			letter := unicode.ToLower(v)
			arr = append(arr, letter)
		}
	}

	for i := range arr {
		first := arr[i]
		second := arr[len(arr)-i-1]
		if first != second {
			return false
		}
	}

	return true
}

func isPalindrome(s string) bool {
	i := 0
	j := len(s) - 1

	arr := []rune(s)

	for i < j {
		l := unicode.ToLower(arr[i])
		r := unicode.ToLower(arr[j])

		if !isLetterOrDigit(l) {
			i++
			continue
		}

		if !isLetterOrDigit(r) {
			j--
			continue
		}

		if l != r {
			return false
		}

		i++
		j--
	}

	return true
}

func isLetterOrDigit(r rune) bool {
	return unicode.IsLetter(r) || unicode.IsDigit(r)
}
